File: burbuja/Burbuja.java
package burbuja;

import java.util.Arrays;

public class Burbuja {

  public static void orderByBurbuja(int[] numbers) {
    int auxiliar;
    for (int i = 0; i < numbers.length; i++) {
      for (int j = i + 1; j < numbers.length; j++) {
        if (numbers[i] > numbers[j]) {
          auxiliar = numbers[i];
          numbers[i] = numbers[j];
          numbers[j] = auxiliar;
        }
      }
    }
    System.out.println(Arrays.toString(numbers));
  }

  public static void main(String[] args) {
    int[] numbers = { 4, 2, 6, 1, 5, 3 };
    orderByBurbuja(numbers);
  }
}

File: burbuja/burbuja.c
#include <stdio.h>
#include <stdlib.h>

int* CreateArray(int cantidad){
  int* numbers = (int*) malloc(cantidad * sizeof(int));
  return numbers;
}

void FreeArray(int* array){
  free(array);
}

void printArray(int numeros[], int cantidad){
  printf("Valores:[ ");
  for(int i = 0; i < cantidad; i++){
    printf("%i ", numeros[i]);
  }
  printf("]");
}

void OrderByBurbuja(int* numbers, int cantEl){
  int auxiliar;

  for(int i = 0; i < cantEl; i++){
    for(int j = i + 1; j < cantEl; j++){
      if(numbers[i] > numbers[j]){
        auxiliar = numbers[i];
        numbers[i] = numbers[j];
        numbers[j] = auxiliar;
      }
    }
  }

  printArray(numbers, cantEl);
}

int main(){
  int cant = 6;
  int* numbers = CreateArray(cant);
  numbers[0] = 6;
  numbers[1] = 4;
  numbers[2] = 2;
  numbers[3] = 1;
  numbers[4] = 5;
  numbers[5] = 3;

  OrderByBurbuja(numbers, cant);
  FreeArray(numbers);
  return 0;
}